plugins/examples/painless-whitelist/build.gradle

```diff
@@ -30,8 +30,12 @@ dependencies {
   compileOnly project(':modules:lang-painless')
 }
 
-integTestCluster {
-  distribution = 'zip'
+if (System.getProperty('tests.distribution') == null) {
```

Comment:
This seems a bit complicated to add to every project. I'm not 100% sure we need to support `-Dtests.distribution=integ-test-zip`. Maybe `integ-test-zip` is a thing we should only use when you don't override the property.

Reply:
It'd be fairly simple to fail the build if someone sets `distribution` to `integ-test-zip`. If we did that then this project could do `distribution = System.getProperty('tests.distribution', 'zip)`.

Reply:
I *might* be cleaner to move the `PluginBuildPlugin` change to a `gradle.projectsEvaluated` and have it handle the logic all in one place.

Reply:
setting distro `integ-test-zip` would offer a way to run some integration tests in a build that would be much quicker as it would skip a lot of heavy lifting. I'm not sure how useful that would be during development, so can't decide if it's worth doing.

Reply:
I don't think that is generally useful. Plugins and modules are the projects which use integ-test-zip, and you can already cd into the plugins or modules dir and run tests from there.

Reply:
Yeah. I don't think we should support setting `-Dtests.distribution=integ-test-zip`. It should be what you get when you run without setting it.